lib/AACStream/aacs.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * mux.js
+ *
+ * Copyright (c) 2016 Brightcove
+ * All rights reserved.
+ *
+ * A stream-based aac to mp4 converter. This utility can be used to
+ * deliver mp4s to a SourceBuffer on platforms that support native
+ * Media Source Extensions.
+ */
+'use strict';
+var Stream = require('../utils/stream.js');
+
+var
+
+/**
+ * Splits an incoming stream of binary data into ADTS and ID3 Frames.
+ */
+
+ AacStream = function() {
+  var
+    everything,
+    receivedTimeStamp = false,
+    timeStamp = 0;
+
+  AacStream.prototype.init.call(this);
+
+  this.setTimestamp = function (frame) {
+    var d = frame.data,
+      size = ((d[3] & 0x01)  << 30) | (d[4]  << 22) | (d[5] << 14) | (d[6] << 6) | d[7] >>> 2;
+    size *= 4;
+    size += d[7] & 0x03;
+    timeStamp = size;
+  };
+
+  this.parseID3 = function(everything) {
+    var returnSize = (everything[6] << 21) | (everything[7] << 14) |
+              (everything[8] << 7) | (everything[9]),
+      flags = everything[5],
+      footerpresent = (flags & 16) >> 4;
+    if (footerpresent) {
+            return returnSize + 20;
+    }
+    return returnSize + 10;
+  };
+
+  this.parseADTS = function(everything) {
+    var lowThree = (everything[5] & 0xE0) >> 5,
+      middle = everything[4] << 3,
+      highTwo = everything[3] & 0x3 << 11;
+      return (highTwo | middle) | lowThree;
+  };
+
+
+  this.push = function(bytes) {
+    var frameSize = 0;
+
+    // If there are bytes remaining from the last segment, prepend them to the
+    // bytes that were pushed in
+    if (everything !== undefined && everything.length) {
+      var tempLength = everything.length;
+      everything = new Uint8Array(bytes.byteLength + tempLength);
+      everything.set(everything.subarray(0, tempLength));
+      everything.set(bytes, tempLength);
+    } else {
+      everything = bytes;
+    }
+
+    while (everything.length >= 10) {
+      if ((everything[0] === 'I'.charCodeAt(0)) &&
+        (everything[1] === 'D'.charCodeAt(0)) &&
+        (everything[2] === '3'.charCodeAt(0))) {
+
+        //check framesize
+        frameSize = this.parseID3(everything);
+
+        //we have enough in the buffer to emit a full packet
+        if(everything.length >= frameSize) {
+          var chunk = {
```

Comment:
Style guide specifies that you can only have one var and it must appear at the top of functions.